### PR TITLE
feat(unlock-app, locksmith, unlock-js) - VAT rate on receipts

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

**VAT disabled** 

Screenshot 2023-05-24 at 12 19 08

**VAT enabled** 

Screenshot 2023-05-25 at 12 18 39

**VAT enabled with validation** 

Screenshot 2023-05-25 at 12 18 28

**Receipt with VAT value** 

Screenshot 2023-05-25 at 12 19 12

**Receipt without VAT value** 

Screenshot 2023-05-25 at 12 19 31

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx [hunk 1]
```diff
@@ -65,18 +80,70 @@ export const ReceiptBaseForm = ({
...
+          <Placeholder.Line size="xl" />
+          <Placeholder.Line size="xl" />
+        </Placeholder.Root>
+        <Placeholder.Line size="xl" />
+        <Placeholder.Line size="xl" />
+        <Placeholder.Line size="xl" />
+        <Placeholder.Line size="xl" />
+        <Placeholder.Line size="xl" />
+        <Placeholder.Line size="xl" />
+      </Placeholder.Root>
```
Comment (searchableguy) [inline unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx:97]: refactor it into a separate component outside this?

File: locksmith/migrations/20230524112140-add-vat-rate-receipt-base.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+'use strict'
+const table = 'ReceiptBases'
+/** @type {import('sequelize-cli').Migration} */
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.addColumn(table, 'vatRatePercentage', {
+      type: Sequelize.INTEGER,
```
Comment (julien51) [inline locksmith/migrations/20230524112140-add-vat-rate-receipt-base.js:8]: This is not good. Some VAT rates might be 8.75%

If you want to use `Sequelize.INTEGER` then we need to represent VAT rates as "basis points" like 2000 for 20% or 875 for 8.75%

Comment (kalidiagne) [reply locksmith/migrations/20230524112140-add-vat-rate-receipt-base.js:8]: renamed as `vatBasisPointsRate`

File: unlock-app/src/components/interface/Receipts/elements/ReceiptBox.tsx [hunk 3]
```diff
@@ -138,11 +140,21 @@ export const ReceiptBox = ({ lockAddress, hash, network }: ReceiptBoxProps) => {
...
+                  <span className="font-semibold">{`${receiptPrice?.total} ${symbol}`}</span>
+                </div>
               </div>
+              {showVatPercentage && (
+                <div className="flex gap-0.5 md:ml-auto">
+                  <span>VAT:</span>
+                  <div className="flex flex-col">
+                    <span className="font-semibold">{`${supplier?.vatRatePercentage} %`}</span>
+                  </div>
+                </div>
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/Receipts/elements/ReceiptBox.tsx:156]: UI changes, screens in the main description

**Pull request merged**